<a href="https://colab.research.google.com/github/vunnampragathi/690-stats-and-visu/blob/main/part%20B/project.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#Project: Exploratory Data Analysis on Environment

#### Name : Pragathi Vunnam

In this project i am doing the environmental data analysis using pandas and plotly express.i have choosen environment data from worlddev.xyz website. There are several indicators to analyse the environment but here i have choosen 3 indicators named Forest area (% of land area),	Total greenhouse gas emissions (kt of CO2 equivalent)	and PM2.5 air pollution, mean annual exposure (micrograms per cubic meter). Here i comparing the relation between forest area , greem house gases and air pollution.

In [38]:
#importing required libraries
import pandas as pd
import plotly.express as px

In [39]:
#reading the data
df = pd.read_csv("https://raw.githubusercontent.com/vunnampragathi/690-stats-and-visu/main/part%20B/wdx_wide_no_NA.csv")

In [40]:
# finding the number of rows and columns present in the data
df.shape

(1486, 10)

it is obsereved that there are 1486 rows and 10 columns are present in the choosen data

In [41]:
# displaying the data
df

Unnamed: 0  Year Country Code          Country Name  \
0              1  2010          AFG           Afghanistan   
1              2  2010          AGO                Angola   
2              3  2010          ALB               Albania   
3              4  2010          AND               Andorra   
4              5  2010          ARE  United Arab Emirates   
...          ...   ...          ...                   ...   
1481        1696  2017          WSM                 Samoa   
1482        1697  2017          YEM           Yemen, Rep.   
1483        1698  2017          ZAF          South Africa   
1484        1699  2017          ZMB                Zambia   
1485        1700  2017          ZWE              Zimbabwe   

                          Region         Income Group    Lending Type  \
0                     South Asia           Low income             IDA   
1            Sub-Saharan Africa   Lower middle income            IBRD   
2          Europe & Central Asia  Upper middle income            IBRD   
3          Europe & Central Asia          High income  Not classified   
4     Middle East & North Africa          High income  Not classified   
...                          ...                  ...             ...   
1481         East Asia & Pacific  Lower middle income             IDA   
1482  Middle East & North Africa           Low income             IDA   
1483         Sub-Saharan Africa   Upper middle income            IBRD   
1484         Sub-Saharan Africa            Low income             IDA   
1485         Sub-Saharan Africa   Lower middle income           Blend   

      AG.LND.FRST.ZS  EN.ATM.GHGT.KT.CE  EN.ATM.PM25.MC.M3  
0           1.852782       27120.000839          65.245592  
1          57.879201       77730.003357          33.787247  
2          28.542701        8899.999619          21.277828  
3          34.042553         629.999995          12.807197  
4           4.467756      201889.999390          38.980068  
...              ...                ...                ...  
1481       58.672662         680.000007          11.548027  
1482        1.039832       23840.000153          50.456007  
1483       14.145109      550700.012207          25.102205  
1484       61.042885       36700.000763          27.438035  
1485       45.451183       28299.999237          22.251671  

[1486 rows x 10 columns]

In [42]:
# from the above data the names of the columns are difficult to understand so i am changing the column names
df = df.rename(columns = {"AG.LND.FRST.ZS":"Forest area (% of land area)","EN.ATM.GHGT.KT.CE":"Total greenhouse gas emissions (kt of CO2 equivalent)","EN.ATM.PM25.MC.M3":"PM2.5 air pollution, mean annual exposure (micrograms per cubic meter)"})
df

Unnamed: 0  Year Country Code          Country Name  \
0              1  2010          AFG           Afghanistan   
1              2  2010          AGO                Angola   
2              3  2010          ALB               Albania   
3              4  2010          AND               Andorra   
4              5  2010          ARE  United Arab Emirates   
...          ...   ...          ...                   ...   
1481        1696  2017          WSM                 Samoa   
1482        1697  2017          YEM           Yemen, Rep.   
1483        1698  2017          ZAF          South Africa   
1484        1699  2017          ZMB                Zambia   
1485        1700  2017          ZWE              Zimbabwe   

                          Region         Income Group    Lending Type  \
0                     South Asia           Low income             IDA   
1            Sub-Saharan Africa   Lower middle income            IBRD   
2          Europe & Central Asia  Upper middle income            IBRD   
3          Europe & Central Asia          High income  Not classified   
4     Middle East & North Africa          High income  Not classified   
...                          ...                  ...             ...   
1481         East Asia & Pacific  Lower middle income             IDA   
1482  Middle East & North Africa           Low income             IDA   
1483         Sub-Saharan Africa   Upper middle income            IBRD   
1484         Sub-Saharan Africa            Low income             IDA   
1485         Sub-Saharan Africa   Lower middle income           Blend   

      Forest area (% of land area)  \
0                         1.852782   
1                        57.879201   
2                        28.542701   
3                        34.042553   
4                         4.467756   
...                            ...   
1481                     58.672662   
1482                      1.039832   
1483                     14.145109   
1484                     61.042885   
1485                     45.451183   

      Total greenhouse gas emissions (kt of CO2 equivalent)  \
0                                          27120.000839       
1                                          77730.003357       
2                                           8899.999619       
3                                            629.999995       
4                                         201889.999390       
...                                                 ...       
1481                                         680.000007       
1482                                       23840.000153       
1483                                      550700.012207       
1484                                       36700.000763       
1485                                       28299.999237       

      PM2.5 air pollution, mean annual exposure (micrograms per cubic meter)  
0                                             65.245592                       
1                                             33.787247                       
2                                             21.277828                       
3                                             12.807197                       
4                                             38.980068                       
...                                                 ...                       
1481                                          11.548027                       
1482                                          50.456007                       
1483                                          25.102205                       
1484                                          27.438035                       
1485                                          22.251671                       

[1486 rows x 10 columns]

In [43]:
# printing top 5 rows of the data
df.head(5)

Unnamed: 0  Year Country Code          Country Name  \
0           1  2010          AFG           Afghanistan   
1           2  2010          AGO                Angola   
2           3  2010          ALB               Albania   
3           4  2010          AND               Andorra   
4           5  2010          ARE  United Arab Emirates   

                       Region         Income Group    Lending Type  \
0                  South Asia           Low income             IDA   
1         Sub-Saharan Africa   Lower middle income            IBRD   
2       Europe & Central Asia  Upper middle income            IBRD   
3       Europe & Central Asia          High income  Not classified   
4  Middle East & North Africa          High income  Not classified   

   Forest area (% of land area)  \
0                      1.852782   
1                     57.879201   
2                     28.542701   
3                     34.042553   
4                      4.467756   

   Total greenhouse gas emissions (kt of CO2 equivalent)  \
0                                       27120.000839       
1                                       77730.003357       
2                                        8899.999619       
3                                         629.999995       
4                                      201889.999390       

   PM2.5 air pollution, mean annual exposure (micrograms per cubic meter)  
0                                          65.245592                       
1                                          33.787247                       
2                                          21.277828                       
3                                          12.807197                       
4                                          38.980068

In [44]:
# displaying last 5 rows of data
df.tail(5)

Unnamed: 0  Year Country Code  Country Name                      Region  \
1481        1696  2017          WSM         Samoa         East Asia & Pacific   
1482        1697  2017          YEM   Yemen, Rep.  Middle East & North Africa   
1483        1698  2017          ZAF  South Africa         Sub-Saharan Africa    
1484        1699  2017          ZMB        Zambia         Sub-Saharan Africa    
1485        1700  2017          ZWE      Zimbabwe         Sub-Saharan Africa    

             Income Group Lending Type  Forest area (% of land area)  \
1481  Lower middle income          IDA                     58.672662   
1482           Low income          IDA                      1.039832   
1483  Upper middle income         IBRD                     14.145109   
1484           Low income          IDA                     61.042885   
1485  Lower middle income        Blend                     45.451183   

      Total greenhouse gas emissions (kt of CO2 equivalent)  \
1481                                         680.000007       
1482                                       23840.000153       
1483                                      550700.012207       
1484                                       36700.000763       
1485                                       28299.999237       

      PM2.5 air pollution, mean annual exposure (micrograms per cubic meter)  
1481                                          11.548027                       
1482                                          50.456007                       
1483                                          25.102205                       
1484                                          27.438035                       
1485                                          22.251671

In [45]:
# displaying random of 5 rows present in the data
df.sample(5)

Unnamed: 0  Year Country Code Country Name                 Region  \
1311        1501  2017          BDI      Burundi    Sub-Saharan Africa    
1190        1360  2016          IDN    Indonesia    East Asia & Pacific   
751          860  2014          AUT      Austria  Europe & Central Asia   
758          867  2014          BGR     Bulgaria  Europe & Central Asia   
1048        1197  2015          MYS     Malaysia    East Asia & Pacific   

             Income Group    Lending Type  Forest area (% of land area)  \
1311           Low income             IDA                     10.889408   
1190  Lower middle income            IBRD                     50.743455   
751           High income  Not classified                     46.989724   
758   Upper middle income            IBRD                     35.130803   
1048  Upper middle income            IBRD                     59.242794   

      Total greenhouse gas emissions (kt of CO2 equivalent)  \
1311                                        4619.999886       
1190                                      847580.017090       
751                                        75150.001526       
758                                        54540.000916       
1048                                      293609.985352       

      PM2.5 air pollution, mean annual exposure (micrograms per cubic meter)  
1311                                          38.896268                       
1190                                          16.384154                       
751                                           13.379228                       
758                                           20.743570                       
1048                                          17.440910

# Analysis

In [46]:
#understanding the basic information about this data
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1486 entries, 0 to 1485
Data columns (total 10 columns):
 #   Column                                                                  Non-Null Count  Dtype  
---  ------                                                                  --------------  -----  
 0   Unnamed: 0                                                              1486 non-null   int64  
 1   Year                                                                    1486 non-null   int64  
 2   Country Code                                                            1486 non-null   object 
 3   Country Name                                                            1486 non-null   object 
 4   Region                                                                  1486 non-null   object 
 5   Income Group                                                            1486 non-null   object 
 6   Lending Type                                                            1486 non

In [47]:
df.shape

(1486, 10)

In [48]:
df.describe()  # getting the statistcal overview of the data

Unnamed: 0         Year  Forest area (% of land area)  \
count  1486.000000  1486.000000                   1486.000000   
mean    848.094886  2013.504038                     32.768438   
std     491.643583     2.290880                     24.076531   
min       1.000000  2010.000000                      0.000000   
25%     425.250000  2012.000000                     11.325622   
50%     850.500000  2014.000000                     31.050935   
75%    1275.500000  2015.750000                     51.269766   
max    1700.000000  2017.000000                     98.076026   

       Total greenhouse gas emissions (kt of CO2 equivalent)  \
count                                       1.486000e+03       
mean                                        2.360911e+05       
std                                         9.966092e+05       
min                                         8.000000e+01       
25%                                         9.242500e+03       
50%                                         3.422000e+04       
75%                                         1.037625e+05       
max                                         1.203527e+07       

       PM2.5 air pollution, mean annual exposure (micrograms per cubic meter)  
count                                        1486.000000                       
mean                                           28.280560                       
std                                            17.658785                       
min                                             5.861331                       
25%                                            15.892197                       
50%                                            23.639767                       
75%                                            35.469170                       
max                                           100.784428

it is difficult to analyse all the data.So i am analysing the data based on particular year.here i am choosing 2010 data for my analysis

In [49]:
df1=df[df['Year']==2010]
df1

Unnamed: 0  Year Country Code          Country Name  \
0             1  2010          AFG           Afghanistan   
1             2  2010          AGO                Angola   
2             3  2010          ALB               Albania   
3             4  2010          AND               Andorra   
4             5  2010          ARE  United Arab Emirates   
..          ...   ...          ...                   ...   
180         205  2010          WSM                 Samoa   
181         206  2010          YEM           Yemen, Rep.   
182         207  2010          ZAF          South Africa   
183         208  2010          ZMB                Zambia   
184         209  2010          ZWE              Zimbabwe   

                         Region         Income Group    Lending Type  \
0                    South Asia           Low income             IDA   
1           Sub-Saharan Africa   Lower middle income            IBRD   
2         Europe & Central Asia  Upper middle income            IBRD   
3         Europe & Central Asia          High income  Not classified   
4    Middle East & North Africa          High income  Not classified   
..                          ...                  ...             ...   
180         East Asia & Pacific  Lower middle income             IDA   
181  Middle East & North Africa           Low income             IDA   
182         Sub-Saharan Africa   Upper middle income            IBRD   
183         Sub-Saharan Africa            Low income             IDA   
184         Sub-Saharan Africa   Lower middle income           Blend   

     Forest area (% of land area)  \
0                        1.852782   
1                       57.879201   
2                       28.542701   
3                       34.042553   
4                        4.467756   
..                            ...   
180                     58.830389   
181                      1.039832   
182                     14.355151   
183                     62.814942   
184                     46.284813   

     Total greenhouse gas emissions (kt of CO2 equivalent)  \
0                                         27120.000839       
1                                         77730.003357       
2                                          8899.999619       
3                                           629.999995       
4                                        201889.999390       
..                                                 ...       
180                                         509.999990       
181                                       52169.998169       
182                                      534119.995117       
183                                       31200.000763       
184                                       28319.999695       

     PM2.5 air pollution, mean annual exposure (micrograms per cubic meter)  
0                                            65.245592                       
1                                            33.787247                       
2                                            21.277828                       
3                                            12.807197                       
4                                            38.980068                       
..                                                 ...                       
180                                          14.288094                       
181                                          45.979470                       
182                                          27.138587                       
183                                          28.836379                       
184                                          23.231479                       

[185 rows x 10 columns]

Even after selecting one year data there are total 185 rows, it is also diffult for analyses so i am choosing some popular countires like china,india,united staes,brazil,pakistan,russia for my further analyses.

In [50]:
df2=df1[df1['Country Name'].isin(['Brazil','China','India','Russian Federation','Pakistan','United States'])]
df2

Unnamed: 0  Year Country Code        Country Name  \
23           26  2010          BRA              Brazil   
32           35  2010          CHN               China   
77           86  2010          IND               India   
130         145  2010          PAK            Pakistan   
141         160  2010          RUS  Russian Federation   
174         197  2010          USA       United States   

                         Region         Income Group    Lending Type  \
23   Latin America & Caribbean   Upper middle income            IBRD   
32          East Asia & Pacific  Upper middle income            IBRD   
77                   South Asia  Lower middle income            IBRD   
130                  South Asia  Lower middle income           Blend   
141       Europe & Central Asia  Upper middle income            IBRD   
174               North America          High income  Not classified   

     Forest area (% of land area)  \
23                      61.207482   
32                      21.285597   
77                      23.374221   
130                      5.310463   
141                     49.773589   
174                     33.749407   

     Total greenhouse gas emissions (kt of CO2 equivalent)  \
23                                        9.914900e+05       
32                                        1.021851e+07       
77                                        2.533990e+06       
130                                       3.223500e+05       
141                                       2.285030e+06       
174                                       6.427130e+06       

     PM2.5 air pollution, mean annual exposure (micrograms per cubic meter)  
23                                           15.955285                       
32                                           69.479570                       
77                                           95.758179                       
130                                          68.005190                       
141                                          19.480311                       
174                                           9.358636

In [51]:
# plotting the bar graph for forest area
px.bar(df2, x = 'Country Name', y = 'Forest area (% of land area)',color='Country Name',title='Bar chart showing the total forest area in selected countries')


from the above bar chart it is obsereved that brazil has highest forest area,it has 60% of total land as forests.russia is in the second place next is united staets, india china and last is pakistan.whereas pakistan has very forest area less than 10% of total land.from the above chart we can say that brazil has good atmospehric conditions compared to the other countries.

In [52]:
# plotting the pie chart for green house gas emissions
px.pie(df2,values='Total greenhouse gas emissions (kt of CO2 equivalent)',color='Country Name',names='Country Name',title='pie chart showing the total green house gases emissions')

from the above pie chart it is obserevd that china has higest green house gas emissions it has approximately about 50% compared to other countries.next highest green house gas emissions is in united states,next is india and pakistan has very less amount of green house gas emissions.from this we can assume that china is the more pollutant country and pakistan is less pollutant compared to other countries.

i want to analyse how the air pollution is changing over the years for the selected countries.so i am selecting the countries from the original dataset

In [53]:
df3=df[df['Country Name'].isin(['Brazil','China','India','Russian Federation','Pakistan','United States'])]
df3

Unnamed: 0  Year Country Code        Country Name  \
23            26  2010          BRA              Brazil   
32            35  2010          CHN               China   
77            86  2010          IND               India   
130          145  2010          PAK            Pakistan   
141          160  2010          RUS  Russian Federation   
174          197  2010          USA       United States   
208          236  2011          BRA              Brazil   
217          245  2011          CHN               China   
262          297  2011          IND               India   
315          357  2011          PAK            Pakistan   
326          372  2011          RUS  Russian Federation   
359          410  2011          USA       United States   
393          449  2012          BRA              Brazil   
402          458  2012          CHN               China   
447          510  2012          IND               India   
500          570  2012          PAK            Pakistan   
511          585  2012          RUS  Russian Federation   
545          623  2012          USA       United States   
579          662  2013          BRA              Brazil   
588          671  2013          CHN               China   
633          723  2013          IND               India   
686          783  2013          PAK            Pakistan   
697          798  2013          RUS  Russian Federation   
731          836  2013          USA       United States   
765          875  2014          BRA              Brazil   
774          884  2014          CHN               China   
819          936  2014          IND               India   
872          996  2014          PAK            Pakistan   
883         1011  2014          RUS  Russian Federation   
917         1049  2014          USA       United States   
951         1088  2015          BRA              Brazil   
960         1097  2015          CHN               China   
1005        1149  2015          IND               India   
1058        1209  2015          PAK            Pakistan   
1069        1224  2015          RUS  Russian Federation   
1103        1262  2015          USA       United States   
1137        1301  2016          BRA              Brazil   
1146        1310  2016          CHN               China   
1191        1362  2016          IND               India   
1244        1422  2016          PAK            Pakistan   
1255        1437  2016          RUS  Russian Federation   
1289        1475  2016          USA       United States   
1323        1514  2017          BRA              Brazil   
1332        1523  2017          CHN               China   
1377        1575  2017          IND               India   
1430        1635  2017          PAK            Pakistan   
1441        1650  2017          RUS  Russian Federation   
1475        1688  2017          USA       United States   

                          Region         Income Group    Lending Type  \
23    Latin America & Caribbean   Upper middle income            IBRD   
32           East Asia & Pacific  Upper middle income            IBRD   
77                    South Asia  Lower middle income            IBRD   
130                   South Asia  Lower middle income           Blend   
141        Europe & Central Asia  Upper middle income            IBRD   
174                North America          High income  Not classified   
208   Latin America & Caribbean   Upper middle income            IBRD   
217          East Asia & Pacific  Upper middle income            IBRD   
262                   South Asia  Lower middle income            IBRD   
315                   South Asia  Lower middle income           Blend   
326        Europe & Central Asia  Upper middle income            IBRD   
359                North America          High income  Not classified   
393   Latin America & Caribbean   Upper middle income            IBRD   
402          East Asia & Pacific  Upper middle income            IBRD   
447                

In [54]:
df3.shape

(48, 10)

In [55]:
# plotting the line chart of air pollution for selected countries from 2010 to 2017
px.line(df3,x='Year',y='PM2.5 air pollution, mean annual exposure (micrograms per cubic meter)',color='Country Name',title='line chart showing annual exposure of air pollution from 2010 to 2017')

from the above line chart it is observed that india is in first place in terms of polluted air,but compared to 2010 the pollution level is reduced in the 2017.china is in second place intially but it pollution levels are decreased as the years passed it is in the 3 rd place in 2017.it seems that USA has very fresh air.

In [56]:
 # plotting the scatter plot for forest area and grrn house gas
 px.scatter(df1,
    x="Total greenhouse gas emissions (kt of CO2 equivalent)",
    y="Forest area (% of land area)", 
    color='Country Name',
    title="scatter plot showing total forest VS total green house gas emissions across all the countries")

from the above scatter plot it is observed that china has highest green house gases emissions compared to all other countries with forest area approximately about 20% of total land.next place is for usa.it seems that as the forest area reduces there is chance of more emmission of green house gases.

In [57]:
# plotting scatter plot for air pollution and forest area
px.scatter(df1,
    x="PM2.5 air pollution, mean annual exposure (micrograms per cubic meter)",
    y="Forest area (% of land area)", 
    color='Country Name',
    title="scatter plot showing total forest VS total air pollution across all the countries")

from the above scatter plot it is obserevd that as the forest area increas the amount of air pollution caused is less.

In [58]:
# plotting scatter plot for green house gas and air pollution
px.scatter(df1,
    x="Total greenhouse gas emissions (kt of CO2 equivalent)",
    y="PM2.5 air pollution, mean annual exposure (micrograms per cubic meter)", 
    color='Country Name',
    title="scatter plot showing total green house gas VS total air pollution across all the countries")

from the above scatter plot it is obserevd that emission of green house gases and air pollution has linear relationship, as the emission of green house gases increases the air pollution is also increasing.

## Conclusion:
Environment plays a crucial role in everyones life.so protecting environment is everyones duty.from the above analysis it is obseverd that forest area has linear relationship with both emission of green house gases and air pollution.to protect our environment we need to increase the forest area and should reduce the use of itmes causing the air pollution and release of greenhouse gases. Because they not only damaging the environment they are very harmful for human health,increase in air pollution and green house gases will result in several health issues for humans.